articles/101_build_tool.md

```diff
@@ -289,59 +291,119 @@ The following items are possible extension points to provide custom functionalit
 - setup the environment (e.g. `sh`, `bash`, `bat`)
 - completion (e.g. `bash`, `Powershell`)
 
-Assuming that the tool will be implemented in Python (since that is the case for existing tools) the entry point mechanism provides a convenient way to make the software extensible.
-Extensions don't have to be integrated into the Python package containing the core logic of the build tool but can easily be provided by additional Python packages.
-This approach will not only foster a modular design and promote clear interfaces but enable external contributions without requiring them to be integrated in a single monolithic package.
-
 ## Possible Approaches
 
-In terms of flexibility neither of the existing build tools can already support the superset of features described in this article.
-There are multiple different paths possible to reach the goal of a universal build tool which fall into two categories:
+When the first draft of this article was written neither of the existing build tools supported the superset of features described in this article.
+There were multiple different paths possible to reach the goal of a universal build tool which fall into two categories:
 
 - One approach is to incrementally evolve one of the existing tools to satisfy the described goals.
 - Another approach would be to start "from scratch".
 
+Since then the new project `colcon` has been developed which covers most of the enumerated requirements and represents the second category.
+
 ### Evolve catkin_make, catkin_make_isolated, or ament_tools
 
 Since neither of these three build tools has the feature richness of `catkin_tools` it is considered strictly less useful to starting building upon one of these build tools.
+Therefore neither of these are being considered as a foundation for a universal build tool.
 
 ### Evolve catkin_tools
 
-Since `catkin_tools` is in many aspects the most complete build tool it should be the one being evolved.
+Since `catkin_tools` is in many aspects the most complete ROS build tool it should be the one being evolved.
 While `ament_tools` has a few features `catkin_tools` currently lacks (e.g. plain CMake support without a manifest, Windows support) the feature richness of `catkin_tools` makes it a better starting point.
 
-The following items are highlighting some of the necessary efforts (not a complete list):
+### Start "from scratch" / colcon
+
+Since the first draft of this article the `colcon` project has been developed with the goals and requirements of a universal build tool in mind.
+In its current form it is already able to build ROS 1 workspaces, ROS 2 workspaces, as well as Gazebo including its ignition dependencies.
```

Comment:
I assume that colcon cannot build workspaces with a mix of ros1 and ros2 packages, since this was a limitation discussed originally. I that true? I did not see it mentioned in the changes or the colcon manuals. Might just help managing expectations to state it explicitly.

Reply:
Should we add a sentence here to highlight the state of colcon regarding support of heterogeneous workspaces?